ui/src/app/subnets/views/SubnetDetails/SubnetSummary/SubnetSummary.tsx

```diff
@@ -1,7 +1,233 @@
+import { useEffect } from "react";
+
+import { Row, Col, Tooltip, Icon } from "@canonical/react-components";
+import { useSelector, useDispatch } from "react-redux";
+
 import TitledSection from "app/base/components/TitledSection";
+import { actions as fabricActions } from "app/store/fabric";
+import fabricSelectors from "app/store/fabric/selectors";
+import { getFabricById } from "app/store/fabric/utils";
+import { actions as spaceActions } from "app/store/space";
+import spaceSelectors from "app/store/space/selectors";
+import { getSpaceById } from "app/store/space/utils";
+import type { Subnet } from "app/store/subnet/types";
+import { actions as vlanActions } from "app/store/vlan";
+import vlanSelectors from "app/store/vlan/selectors";
+import { getVlanById } from "app/store/vlan/utils";
+
+type Props = {
+  subnet: Subnet;
```

Comment:
We generally prefer to pass ids around rather than full objects. The general rule being that if it's in the store then the data should be fetched with a selector. This makes it easier to understand what data the component is using and avoids unnecessary cascading rerenders if the subnet changes (each component will rerender when the specific data it selects changes).